lib/puppetserver/ca/clean_action.rb

```diff
@@ -0,0 +1,154 @@
+require 'puppetserver/ca/utils'
+require 'puppetserver/utils/http_utilities'
+require 'puppetserver/utils/file_utilities'
+require 'puppetserver/ca/puppet_config'
+require 'puppetserver/ca/revoke_action'
+
+require 'optparse'
+require 'json'
+
+module Puppetserver
+  module Ca
+    class CleanAction
+
+      include Puppetserver::Utils
+
+      CERTNAME_BLACKLIST = %w{--all --config}
+
+      SUMMARY = 'Clean files from the CA for certificate(s)'
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca clean [--help|--version]
+  puppetserver ca clean [--config] --certname CERTNAME[,ADDLCERTNAME]
+
+Description:
+Given one or more valid certnames, instructs the CA to remove files
+pertaining to them (keys, cert, certificate request) over HTTPS using
+the local agent's PKI
```

Comment:
This also revokes the certificate that was issued, I think we should mention that. I found the previous docs we had confusing because they didn't explain how `clean` interacted with `revoke`.

Reply:
Updated this, let me know if it still needs improvement!